### PR TITLE
Refratorization

### PR DESCRIPTION
**Pull request by vladbatalan**

### CODE REVIEW

File: BatalanVlad/src/PaooGame/Physics/PVector.java [hunk 1]
```diff
@@ -1,46 +1,103 @@
...
-import java.awt.*;
+/**
+ * PVector is a 2D vector.
+ * It can be interpreted in at least two ways:
+ *  - As a Force vector: Velocity, Acceleration, Resultant Force
+ *  - As a Point on the map: the posision Vector.
+ *
+ * @author      Batalan Vlad
+ * @since 1.0
+ */
```
Comment (invo12) [inline BatalanVlad/src/PaooGame/Physics/PVector.java]: Class comments should be like this:
Class responsible for reprenting a 2D vector.

That's all, no explanations given

File: BatalanVlad/src/PaooGame/Physics/PVector.java [hunk 2]
```diff
@@ -1,46 +1,103 @@
...
 
 public class PVector {
+
+    /**
+     * The x coordonate of the PVector.
+     */
     private float x;
```
Comment (invo12) [inline BatalanVlad/src/PaooGame/Physics/PVector.java]: No comments needed for fields, just for public functions

File: BatalanVlad/src/PaooGame/Physics/PVector.java [hunk 3]
```diff
@@ -1,46 +1,103 @@
...
+
+    /**
+     * The y coordonate of the PVector.
+     */
     private float y;
 
+    /**
+     * Default constructor. All the coordonates are initialised with 0.
+     */
     public PVector(){
```
Comment (invo12) [inline BatalanVlad/src/PaooGame/Physics/PVector.java]: Same ideea, no need to comment constructors, getters and setters, it's common sense

File: BatalanVlad/src/PaooGame/Physics/PVector.java [hunk 4]
```diff
@@ -1,46 +1,103 @@
...
     }
 
+    /**
+     * The clone function is used to create a copy of the PVector.
+     * @return
+     *  A clone of the object.
+     * @throws CloneNotSupportedException
+     *  If clonning is not supported
+     */
     @Override
```
Comment (invo12) [inline BatalanVlad/src/PaooGame/Physics/PVector.java]: Be short and on point. Maybe something like this:
Function responsible for creating a copy of the PVector.
Try to leave a newline between first comment and the rest.
Example:
"Text

@return..

File: BatalanVlad/src/PaooGame/Physics/PVector.java [hunk 5]
```diff
@@ -1,46 +1,103 @@
...
+ * It can be interpreted in at least two ways:
+ *  - As a Force vector: Velocity, Acceleration, Resultant Force
+ *  - As a Point on the map: the posision Vector.
+ *
+ * @author      Batalan Vlad
+ * @since 1.0
+ */
 
 public class PVector {
+
```
Comment (invo12) [inline BatalanVlad/src/PaooGame/Physics/PVector.java]: What does Pvector stands for, should refactor name to be easy to read. If P stands for point refactor something like PointVector, or something like this

File: BatalanVlad/src/PaooGame/Tiles/Map.java [hunk 6]
```diff
@@ -119,7 +113,7 @@ public void Draw(Graphics g){
          */
     }
 
-    public void MapExtend(){
+    public void mapExtend(){
         if(map_height < MIN_HEIGHT){
```
Comment (invo12) [inline BatalanVlad/src/PaooGame/Tiles/Map.java:117]: Try to leave a blank line between function name and instructions, looks nicer. Also if you have for loops,if,while, try to leave a blank line asswell if the body has more than 1 line.

File: BatalanVlad/src/PaooGame/Game.java [hunk 7]
```diff
@@ -15,26 +15,67 @@
 import java.awt.image.BufferStrategy;
 import java.util.ArrayList;
 
+
+/**
+ * The main instance of the game
+ */
 public class Game implements Runnable
```
Comment (invo12) [inline BatalanVlad/src/PaooGame/Game.java:21]: Class responsible for running an instance of the game. All class comments should start with "class responsible for"

File: BatalanVlad/src/PaooGame/Game.java [hunk 8]
```diff
@@ -15,26 +15,67 @@
...
 {
-    private GameWindow      wnd;        /*!< Fereastra in care se va desena tabla jocului*/
-    private boolean         runState;   /*!< Flag ce starea firului de executie.*/
+    /**
+     * The main window of the game.
+     */
+    private GameWindow      wnd;
+
```
Comment (invo12) [inline BatalanVlad/src/PaooGame/Game.java:27]: Make names suggestive, you have recommandations from IDE, use them, rename wnd to gameWindow. Make code easy to read

File: BatalanVlad/src/PaooGame/Game.java [hunk 9]
```diff
@@ -134,7 +170,7 @@ public synchronized void StartGame()
      */
     public synchronized void StopGame()
     {
-        if(runState == true)
+        if(runState)
         {
                 /// Actualizare stare thread
             runState = false;
```
Comment (invo12) [inline BatalanVlad/src/PaooGame/Game.java:161]: Don't over comment. If you need comments to explain something that is not a class/public function you have one of the following situations:
1) that piece of code is hard and it needs clarifications
2) your code is not expressive enough, and you need refactorization

If you are not in 1) then you don't need to comment that stuff. Keep that in mind.

File: BatalanVlad/src/PaooGame/Physics/Body.java [hunk 10]
```diff
@@ -127,21 +131,18 @@ public void ajustPositionOnCollision(){
...
 
-        // System.out.println(resultantForce.toString());
         position = oldPosition.add(resultantForce);
-        //System.out.println(collisionState[0] + " " + collisionState[1] + " " + collisionState[2] + " " + collisionState[3] + " " + collisionState[4]);
-        //System.out.println("oldPosition: "+ oldPosition.toString() + " - newPosition: " + position.toString());
     }
 
     public void Stand(){
         //if only one command is running or none
-        if(actions[0] == false && actions[1] == false)
```
Comment (invo12) [inline BatalanVlad/src/PaooGame/Physics/Body.java:138]: Okay, so if ever, the order of actions change, your game is done for life. Use an enum better, or a mapping function between the array and the enum. But this is not necesarry for this commit, but next ones.

File: BatalanVlad/src/PaooGame/Physics/PVector.java (hunk 1 above)
Comment (vladbatalan) [reply BatalanVlad/src/PaooGame/Physics/PVector.java]: Done

File: BatalanVlad/src/PaooGame/Physics/PVector.java (hunk 2 above)
Comment (vladbatalan) [reply BatalanVlad/src/PaooGame/Physics/PVector.java]: Done

File: BatalanVlad/src/PaooGame/Physics/PVector.java (hunk 3 above)
Comment (vladbatalan) [reply BatalanVlad/src/PaooGame/Physics/PVector.java]: Solved

File: BatalanVlad/src/PaooGame/Physics/PVector.java (hunk 4 above)
Comment (vladbatalan) [reply BatalanVlad/src/PaooGame/Physics/PVector.java]: Done

File: BatalanVlad/src/PaooGame/Physics/PVector.java (hunk 5 above)
Comment (vladbatalan) [reply BatalanVlad/src/PaooGame/Physics/PVector.java]: Refactored to PointVector

File: BatalanVlad/src/PaooGame/Tiles/Map.java (hunk 6 above)
Comment (vladbatalan) [reply BatalanVlad/src/PaooGame/Tiles/Map.java:117]: Tried to do this for every occasion

File: BatalanVlad/src/PaooGame/Game.java (hunk 7 above)
Comment (vladbatalan) [reply BatalanVlad/src/PaooGame/Game.java:21]: I left this as it is and I'll try to use for the next ones

File: BatalanVlad/src/PaooGame/Game.java (hunk 8 above)
Comment (vladbatalan) [reply BatalanVlad/src/PaooGame/Game.java:27]: Refactored to gameWindow

File: BatalanVlad/src/PaooGame/Game.java (hunk 9 above)
Comment (vladbatalan) [reply BatalanVlad/src/PaooGame/Game.java:161]: Made the simplifications

**Pull request merged**